### PR TITLE
add block usage logs to system pallet

### PR DESCRIPTION
**Pull request by kianenigma**
not sure if a good idea, I needed them for some testing, and figured it might be good to have around. The code block is a bit ugly though.

### CODE REVIEW

File: frame/system/src/lib.rs [hunk 1]
```diff
@@ -1272,6 +1272,40 @@ impl<T: Config> Pallet<T> {
 	/// Remove temporary "environment" entries in storage, compute the storage root and return the
 	/// resulting header for this block.
 	pub fn finalize() -> T::Header {
+		log::debug!(
+			target: "runtime::system",
+			"[{:?}] length: {} (normal {:?}, op: {:?}, mandatory {:?}) / normal weight: {} ({:?}) \
```
Comment (bkchr) [inline frame/system/src/lib.rs:1277]:
```suggestion
			"[{:?}] ext num: {} (normal {:?}, op: {:?}, mandatory {:?}) / normal weight: {} ({:?}) \
```

**Review by bkchr — COMMENTED** [review]
Could you show an example output?

**Comment by kianenigma** [comment]
`2022-03-02 22:27:48.001 DEBUG tokio-runtime-worker runtime::system: [11] length: 11 (normal Percent(0), op: Percent(0), mandatory Percent(0)) / normal weight: 0 (Percent(0)) / op weight 0 (Percent(0)) / mandatory weight 5795614000 (Percent(0))`

**Comment by kianenigma** [comment]
Of course, this is a rather lame example.

File: frame/system/src/lib.rs [hunk 2]
```diff
@@ -1272,6 +1272,40 @@ impl<T: Config> Pallet<T> {
...
+			/ op weight {} ({:?}) / mandatory weight {} ({:?})",
+			Self::block_number(),
+			Self::all_extrinsics_len(),
+			sp_runtime::Percent::from_rational(
+				Self::all_extrinsics_len(),
+				*T::BlockLength::get().max.get(DispatchClass::Normal)
+			),
```
Comment (bkchr) [inline frame/system/src/lib.rs:1284]: What should this express?

**Comment by bkchr** [comment]
> `2022-03-02 22:27:48.001 DEBUG tokio-runtime-worker runtime::system: [11] length: 11 (normal Percent(0), op: Percent(0), mandatory Percent(0)) / normal weight: 0 (Percent(0)) / op weight 0 (Percent(0)) / mandatory weight 5795614000 (Percent(0))`

Quite hard to read? :see_no_evil:

File: frame/system/src/lib.rs (hunk 2 above)
Comment (kianenigma) [reply frame/system/src/lib.rs:1284]: What portion of the block that's available to Normal extrinsics' length is full.

File: frame/system/src/lib.rs (hunk 1 above)
Comment (kianenigma) [reply frame/system/src/lib.rs:1277]: hmm no, this is extrinsic length, not count.

**Comment by kianenigma** [comment]
> > `2022-03-02 22:27:48.001 DEBUG tokio-runtime-worker runtime::system: [11] length: 11 (normal Percent(0), op: Percent(0), mandatory Percent(0)) / normal weight: 0 (Percent(0)) / op weight 0 (Percent(0)) / mandatory weight 5795614000 (Percent(0))`
> 
> Quite hard to read? 🙈

I can maybe show less information, but I think this is fine. I mean, it is a _debug_ log at the end of the day, not meant to be pretty to read :P

File: frame/system/src/lib.rs (hunk 2 above)
Comment (bkchr) [reply frame/system/src/lib.rs:1284]: I'm an idiot :P

File: frame/system/src/lib.rs (hunk 1 above)
Comment (bkchr) [reply frame/system/src/lib.rs:1277]: :facepalm:

**Review by bkchr — APPROVED** [review]
I still find the output confusing, even for debug output :P

**Review by ggwpez — APPROVED** [review]
Wow that is a lot of code for one print!  
Maybe need to start putting it it's own func.  
Could you post another output of how it looks like now?

**Comment by kianenigma** [comment]
> Wow that is a lot of code for one print! Maybe need to start putting it it's own func. Could you post another output of how it looks like now?

> Wow that is a lot of code for one print! Maybe need to start putting it it's own func. Could you post another output of how it looks like now?

Good idea!

**Comment by kianenigma** [comment]
I changed it to something like this: 

```
2022-03-04 21:19:12.004 DEBUG tokio-runtime-worker runtime::system: [922] length: 11 (normal 0%, op: 0%, mandatory 0%) / normal weight: 0 (0%) / op weight 0 (0%) / mandatory weight 7464066000 (0%)
```

**Comment by kianenigma** [comment]
bot merge

**Pull request merged**